### DISCUSSION TITLE
sb.type()，Cannot input to div，In an element that is not a text type

### DISCUSSION THREAD

**Discussion by luckyboy-wei** [seleniumbase/SeleniumBase#2703] [Q&A] [answered]
![image](https://github.com/seleniumbase/SeleniumBase/assets/49641020/ef9b0dd9-81ae-4c7e-ba4c-90704232be86)

sb.type('//div[@data-mode-id="sql"]','select * from t_transfer')

Unable to enter

**Comment by mdmintz** [comment, marked as answer]
`sb.type(selector, text)` only works on `input` fields. (Same with regular Selenium's `send_keys()` method.)

You could try using other methods though, like `sb.js_type(selector, text)` or `sb.set_text_content(selector, text)`. Maybe even raw JS if those don't work for you.

**Reply by luckyboy-wei** [reply]
sb.js_type(selector, text) or sb.set_text_content(selector, text)

None of the above methods can be realized. There are other methods

**Reply by mdmintz** [reply]
All methods: [SeleniumBase/help_docs/method_summary.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/method_summary.md)
